array.go

```diff
@@ -482,30 +540,232 @@ func newArrayDataSlabFromDataV1(
 		header:    header,
 		elements:  elements,
 		extraData: extraData,
+		inlined:   false, // this function is only called when slab is not inlined.
 	}, nil
 }
 
-// Encode encodes this array data slab to the given encoder.
+// DecodeInlinedArrayStorable decodes inlined array data slab. Encoding is
+// version 1 with CBOR tag having tag number CBORTagInlinedArray, and tag contant
+// as 3-element array:
 //
-// Root DataSlab Header:
+//	+------------------+----------------+----------+
+//	| extra data index | value ID index | elements |
+//	+------------------+----------------+----------+
+//
+// NOTE: This function doesn't decode tag number because tag number is decoded
+// in the caller and decoder only contains tag content.
+func DecodeInlinedArrayStorable(
+	dec *cbor.StreamDecoder,
+	decodeStorable StorableDecoder,
+	parentSlabID SlabID,
+	inlinedExtraData []ExtraData,
+) (
+	Storable,
+	error,
+) {
+	const inlinedArrayDataSlabArrayCount = 3
+
+	arrayCount, err := dec.DecodeArrayHead()
+	if err != nil {
+		return nil, NewDecodingError(err)
+	}
+
+	if arrayCount != inlinedArrayDataSlabArrayCount {
+		return nil, NewDecodingError(
+			fmt.Errorf(
+				"failed to decode inlined array data slab: expect %d elements, got %d elements",
+				inlinedArrayDataSlabArrayCount,
+				arrayCount))
+	}
+
+	// element 0: extra data index
+	extraDataIndex, err := dec.DecodeUint64()
+	if err != nil {
+		return nil, NewDecodingError(err)
+	}
+	if extraDataIndex >= uint64(len(inlinedExtraData)) {
+		return nil, NewDecodingError(
+			fmt.Errorf(
+				"failed to decode inlined array data slab: inlined extra data index %d exceeds number of inlined extra data %d",
+				extraDataIndex,
+				len(inlinedExtraData)))
+	}
+
+	extraData, ok := inlinedExtraData[extraDataIndex].(*ArrayExtraData)
+	if !ok {
+		return nil, NewDecodingError(
+			fmt.Errorf(
+				"failed to decode inlined array data slab: expect *ArrayExtraData, got %T",
+				inlinedExtraData[extraDataIndex]))
+	}
+
+	// element 1: slab index
+	b, err := dec.DecodeBytes()
+	if err != nil {
+		return nil, NewDecodingError(err)
+	}
+	if len(b) != slabIndexSize {
+		return nil, NewDecodingError(
+			fmt.Errorf(
+				"failed to decode inlined array data slab: expect %d bytes for slab index, got %d bytes",
+				slabIndexSize,
+				len(b)))
+	}
+
+	var index [8]byte
```

Comment:
Maybe define a type like `type Index [8]byte`, and use it here:
```suggestion
	var index Index
```

See also the comment below

Reply:
Done (slightly different) in commit 90be7ac4d23abbaa3128dbd076aa2338a9b12ae5.

I used `SlabIndex` instead of creating a new type.